modules/features/profile/src/main/java/au/com/shiftyjelly/pocketcasts/profile/AccountDetailsFragment.kt

```diff
@@ -242,11 +255,48 @@ class AccountDetailsFragment : BaseFragment() {
         val builder = AlertDialog.Builder(themedContext)
         builder.setTitle(getString(LR.string.profile_sign_out))
             .setMessage(getString(LR.string.profile_sign_out_confirm))
-            .setPositiveButton(getString(LR.string.profile_sign_out)) { _, _ -> performSignOut() }
+            .setPositiveButton(getString(LR.string.profile_sign_out)) { _, _ -> clearDataAlert() }
             .setNegativeButton(getString(LR.string.cancel), null)
             .show()
     }
 
+    private fun clearDataAlert() {
+        val context = context ?: return
+        val themedContext = if (Util.isAutomotive(context)) ContextThemeWrapper(context, CR.style.Theme_Car_NoActionBar) else context
+        val builder = AlertDialog.Builder(themedContext)
+        builder.setTitle(getString(LR.string.profile_clear_data_question))
+            .setMessage("Would you also like to clear all data from the app?")
+            .setPositiveButton("Just sign out") { _, _ -> performSignOut() }
+            .setNegativeButton(getString(LR.string.profile_clear_data)) { _, _ ->
+                signOutAndClearData()
+            }
+            .show()
+    }
+
+    private fun signOutAndClearData() {
+        // Sign out first to make sure no data changes get synced
+        userManager.signOut(playbackManager, wasInitiatedByUser = true)
+
+        // Block while clearing data so that we don't return to the app until the users data has been cleared
+        runBlocking(Dispatchers.IO) {
+            upNextQueue.removeAll()
+
+            playlistManager.resetDb()
+            folderManager.deleteAll()
+            searchHistoryManager.clearAll()
+
+            podcastManager.deleteAllPodcasts()
+
+            userEpisodeManager.findUserEpisodes().forEach {
+                userEpisodeManager.delete(it, playbackManager)
+            }
+            episodeManager.deleteDownloadedEpisodeFiles()
+            episodeManager.deleteAll()
+        }
```

Comment:
I could have also called `SearchHistoryManagerImpl::clearAll` here, but it seemed unnecessary because I don't think we are using search history in Android Automotive. Let me know if I'm wrong about that or if you think we should clear the search history just to have everything covered (I waffled back and forth on whether to do that).

Reply:
Another thing I'm not clearing is the `up_next_changes` table because I assume we would still want to sync any pending Up Next changes back to the account, even if the user has signed out.

Reply:
> I could have also called SearchHistoryManagerImpl::clearAll here, but it seemed unnecessary

I don't believe the Automotive UI supports this so it's fine to leave it out.

> I'm not clearing is the up_next_changes

I think we should clear the Up Next changes as the user can't call the server once we have removed their account from the Account Manager.

Reply:
> I think we should clear the Up Next changes as the user can't call the server once we have removed their account from the Account Manager.

Makes sense. Did that update in [8cc15b3](https://github.com/Automattic/pocket-casts-android/pull/828/commits/8cc15b3ac4a5b1cd577a1a7364fe650a73b98bc0).

This is making me wonder, do you think we should be forcing a sync before logging out?

Reply:
I reckon in the first version we shouldn't worry, it will add another level of complexity. Thanks for all these changes, I'm going to merge it in.